### ISSUE TITLE
Change minimum items

### ISSUE THREAD

**Issue by stansmith907** [adiwg/mdJson-schemas#131]
Change the minimum numbers of items allowed in non-required arrays and objects to zero.

- [x] citation > date
- [x] citation > responsibleParty
- [x] citation > series 
- [ ] ~~constraint > useLimitation is not required for any constraint type~~
- [x] constraint > security and legal are not required unless requested by 'type'.  e.g. type='security' requires security{} but not legal{}.  This may require changes to legalConstraint and/or securityConstraint schema in addition to constraint.
- [x] georeferenceable > orientationParameterDescription (see comment below)
- [ ] ~~metadataInfo > resourceScope missing? see comment below~~
- [ ] ~~lineage < scope is optional but not allowed to be empty object~~
- [ ] ~~mdIdentifier < authority does not allow empty object~~
- [x] measure < type enum should be [distance|length|angle|measure|scale] see comment below
- [ ] ~~medium < mediumSpecification is not required and is not allowed to be empty~~
- [ ] ~~medium < identifier is not required and is not allowed to be empty~~
- [ ] ~~metadataInfo < metadataDate is not required and not allowed to be empty~~
- [ ] ~~orderProcess < plannedAvailability is not required and not allowed to be empty~~
- [ ] ~~processStep < timePeriod is not required and not allowed to be empty~~
- [ ] ~~spatialReferenceSystem < referenceSystemType is not required and not allowed to be empty~~
- [ ] ~~spatialReferenceSystem < referenceSystemIdentifier is not required and not allowed to be empty~~
- [x] taxonomy < voucher  is not required but array is not allowed to be empty
- [x] timeInterval could use come restrictions (see comment below)

**Comment by jlblcc** [comment]
>  constraint > security and legal are not required unless requested by 'type'. e.g. type='security' requires security{} but not legal{}. This may require changes to legalConstraint and/or securityConstraint schema in addition to constraint.

This is already implemented.

**Comment by stansmith907** [comment]
I kind of agree that we should require 'useLimitation' for type='use'.  I'll adjust the code and tests.

**Comment by stansmith907** [comment]
georeferenceable > orientationParameterDescription is not a required field, however if present a length of at least one character is being enforces.  Therefore "orientationParameterDescription": ""  is  not allowed.  Although this makes some sense it's not how we are handling other non-required text fields.

**Comment by stansmith907** [comment]
Refresh my mind.  We had some discussion about restructuring scope #100 and the ability to use it populate hierarchy in metadataInfo #74.  Looking at our issues it seems we decided to add a "resourceScope" array to metadataInfo.  Except neither of us did - not in my mdJson reader modules or mdJson test scripts, not in your schema.  So, an oversight?  Or did we decide something else entirely?

**Comment by jlblcc** [comment]
I think we settled on [resourceType ](http://mdtools.adiwg.org/#viewer-page?v=2-1-0)instead and dropped scope from metadataInfo entirely.

**Comment by stansmith907** [comment]
That sounds familiar.  I'll clean up the metadataInfo model.

**Comment by stansmith907** [comment]
I think our measure enum is off.  We have [distance|length|vertical|angle] this does not validate in either -2 or -1.  What does validate is  [distance|length|angle|measure|scale].  Not sure where we got 'vertical'.  We might have left out 'measure' and 'scale' because all types are not used in each usage of measure.  Let me know if you are good with removing 'vertical'.

**Comment by jlblcc** [comment]
> lineage < scope is optional but not allowed to be empty object

Scope cannot be empty because scopeCode is required

**Comment by jlblcc** [comment]
> mdIdentifier < authority does not allow empty object

authority is a citation and requires a title

**Comment by jlblcc** [comment]
>  medium < mediumSpecification is not required and is not allowed to be empty
> medium < identifier is not required and is not allowed to be empty

Both of those objects have required fields and cannot be empty (citation and identifier).

**Comment by jlblcc** [comment]
> metadataInfo < metadataDate is not required and not allowed to be empty

dateObject has required fields and cannot be empty

BTW, none of the schemas referenced here will allow a `null` value in place of an object.

**Comment by stansmith907** [comment]
Can you add some restrictions to timeInterval?  "units" can only be [ year|month|day|hour\Minute|second ].  Also, "interval" must be an integer (not real or alpha).  Thanks.

**Comment by stansmith907** [comment]
Sorry, messed up last request.  The temporal extensions "duration" and "timeInterval" are GML not ISO so finding documentation has taken some extra searching.   I found some NOAA documentation with more explanation.  "timeInterval" < "interval" values can be real - not integer as I requested in the previous comment.  Also, "duration" values must integer ( not real or alpha ).  

Also got some better definitions for "timeInterval" and "duration" which I will move into "timePeriod".
